examples/simulation/plot_simulated_sample_dataset.py

```diff
@@ -0,0 +1,182 @@
+                         for activation in activation_list))
+region_names = list(activations.keys())
+
+#  Define the time course of the activity for each region to activate. We use a
+#  sine wave and it will be the same for all 4 regions.
+source_time_series = np.sin(np.linspace(0, 4 * np.pi, 100)) * 10e-9
+
+
+# Create simulated source activity.
+
+# Here, SourceSimulator is used, which allows to specifiy where (label), what
+# (source_time_series), and when (events) event type will occur.
+
+# We will add data for 4 areas, each of which contains 2 labels. Since add_data
+# method accepts 1 label per call, it will be called 2 times per area.
+# All activations will contain the same waveform, but the amplitude will be 2
+# times higher in the contralateral label, as explained before.
+
+# When the activity occurs is defined using events. In this case, they are taken
+# from the original raw data. The first column is the sample of the event, the
+# second is not used. The third one is the event id, which is different for each
+# of the 4 areas.
+
+region_id = 0
+source_simulator = mne.simulation.SourceSimulator(src, tstep=tstep)
+for i, region_name in enumerate(region_names):
+    region_id = i+1
+    events_tmp = events[np.where(events[:, 2] == region_id)[0], :]
+    for j in range(2):
+        label_name = activations[region_name][j][0]
+        label_tmp = mne.read_labels_from_annot(subject, annot,
+                                               subjects_dir=subjects_dir,
+                                               regexp=label_name,verbose=False)
+        label_tmp = label_tmp[0]
+        amplitude_tmp = activations[region_name][j][1]
+        source_simulator.add_data(label_tmp, amplitude_tmp*source_time_series,
+                                  events_tmp)
+
+# To obtain a SourceEstimate object, we need to use `get_stc()` method of
+# SourceSimulator class.
+stc_data = source_simulator.get_stc()
+
+
+# Simulate raw data.
+
+# Project the source time series to sensor space. Three types of noise will be
+# added to the simulated raw data:
+#     - multivariate Gaussian noise computed using a noise covariance from
+#       original epoched data
+#     - blink (EOG) noise
+#     - ECG noise
+# The source simulator can be given directly to the `simulate_raw` function.
+
+raw_sim = mne.simulation.simulate_raw(info, source_simulator, forward=fwd,
+                                      cov=None)
+noise_epochs = mne.Epochs(raw, events, tmax=0)
+noise_epochs.info['bads'] = []
+noise_cov = mne.compute_covariance(noise_epochs)
+
+mne.simulation.add_noise(raw_sim, cov=noise_cov, random_state=0)
+mne.simulation.add_eog(raw_sim, random_state=0)
+mne.simulation.add_ecg(raw_sim, random_state=0)
+
+# Plot original and simulated raw data.
+raw.plot()
+raw_sim.plot(title='Simulated raw data from sample dataset')
+
+
+# Reconstruct simulated source time courses using dSPM inverse operator.
```

Comment:
Again line of `####` above and `----` below and keep `#` lines constant here to render this block as RST